### ISSUE TITLE
Compilation fails with target `wasm32-unknown-emscripten`

### ISSUE THREAD

**Issue by Elzapat** [bevyengine/bevy#4150]
## Bevy version
Bevy 0.6.1

## Operating system & version
Manjaro Linux 21.2.4

## What you did
Trying to compile any cargo project with the bevy dependency and with target `wasm32-unknown-emscripten`.
The error happens in my middle-sized personal project and in a new cargo project with `Bevy 0.6.1` as its only dependency.

## What you expected to happen
Successful compilation.

## What actually happened
Compilation failed with two errors, one arising in crate `stdweb` and the other in crate `wgpu-hal`:
```test
error: recursion limit reached while expanding `_js_impl!`
   --> /home/morgan/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/stdweb-0.1.3/src/webcore/macros.rs:124:9
    |
124 |           _js_impl!( @_stringify $arg_counter [")" $($terminator)*] [$($out)* ("(")] -> [$($inner)*] [$($remaining)*] $($rest)* )
    |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
   ::: /home/morgan/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/stdweb-0.1.3/src/webcore/initialization.rs:17:5
    |
17  | /     js! { @(no_return)
18  | |         Module.STDWEB = {};
19  | |         Module.STDWEB.to_js = function to_js( address ) {
20  | |             var kind = HEAPU8[ address + 12 ];
...   |
81  | |         };
82  | |     };
    | |_____- in this macro invocation
    |
    = help: consider increasing the recursion limit by adding a `#![recursion_limit = "1500"]` attribute to your crate (`stdweb`)
    = note: this error originates in the macro `_js_impl` (in Nightly builds, run with -Z macro-backtrace for more info)
```
```text
error[E0599]: no function or associated item named `from_webgl2_context` found for struct `glow::Context` in the current scope
  --> /home/morgan/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/wgpu-hal-0.12.4/src/gles/web.rs:54:32
   |
54 |                 glow::Context::from_webgl2_context(webgl2_context)
   |                                ^^^^^^^^^^^^^^^^^^^ function or associated item not found in `glow::Context`
```

## Additional information
For the `stdweb` error, I have tried raising the recursion limit with `#![recursion_limit = "1500"]` at the top of my `main.rs` but it didn't fix the error, I have tried with really big values as well.
I have tried compiling `wgpu-hal` on its own, in a new cargo project, with target `wasm32-unknown-emscripten` and it compiles successfully, that is why I'm posting the issue here instead of on the `wgpu` repo.

**Comment by bjorn3** [comment]
Using wgpu with emscripten requires enabling the emscripten feature here: https://github.com/bevyengine/bevy/blob/2b112026147a8b6780759b32f0ea04dc7d7d8548/crates/bevy_render/Cargo.toml#L41

While stdweb should work with emscripten, stdweb seems to be no longer maintained. Something probably changed in the mean time that causes it to no longer compile for emscripten. Stdweb is used by gilrs and cpal. Gilrs is used for joystick input. You can disable the bevy_gilrs feature to prevent compilation of it: https://github.com/bevyengine/bevy/blob/2b112026147a8b6780759b32f0ea04dc7d7d8548/Cargo.toml#L21 Cpal is used for audio. You can disable the bevy_audio feature to prevent compilation of it: https://github.com/bevyengine/bevy/blob/2b112026147a8b6780759b32f0ea04dc7d7d8548/Cargo.toml#L20

**Comment by Elzapat** [comment]
Thanks a lot for the help, it pointed me in the right direction.

I managed to compile my project, but I had to make a few adjustments:

After removing the `bevy_gilrs` and `bevy_audio` features from bevy, `stdweb` was still being compiled, so by trial and error I found that I also had to remove the `vorbis` feature for stdweb to disappear, so the remaining features I have are those:
```toml
[dependencies.bevy]
version = "0.6.1"
default-features = false
features = [
  "bevy_winit",
  "render",
  "png",
  "hdr",
  "x11",
  "filesystem_watcher",
]
```

As for `wgpu`, the `emcripten` feature seems to not exist in version `0.12` so I had to override the dependency to the `master` branch of `wgpu` for all my depencies likes this:
```toml
[replace]
"wgpu:0.12.0" = { git = "https://github.com/gfx-rs/wgpu" }
```

And it compiles for me! Thanks again for the help.
